src/Service/Notification.php

```diff
@@ -0,0 +1,64 @@
+<?php declare(strict_types=1);
+
+namespace SilverStripe\MFA\Service;
+
+use Exception;
+use Psr\Log\LoggerInterface;
+use SilverStripe\Control\Email\Email;
+use SilverStripe\Core\Extensible;
+use SilverStripe\Core\Injector\Injectable;
+use SilverStripe\Core\Injector\Injector;
+use SilverStripe\Security\Member;
+
+/**
+ * Ecapsulates setting up an Email in order to allow for Dependency Injection and to avoid introducing a hard
+ * coupling to the SilverStripe core Email class in code that consumes this class.
+ */
+class Notification
+{
+    use Injectable;
+    use Extensible;
+
+    /**
+     * @var Email
+     */
+    private $email;
+
+    /**
+     * Builds the notification delivery class & adds details to it. This is by default an {@see Email} as built by
+     * Injector. Data can contain 'from', 'to', and 'subject' keys, which will be set on the Email before sending.
+     * A caveat to be aware of is that setting 'to' will obviously override the given Member as the recipient.
+     *
+     * @param Member $member Member the notification is being sent to
+     * @param string $template Name of the HTMLTemplate to use for the email body
+     * @param array $data (optional) Extra data for use in populating the template
+     */
+    public function __construct(Member $member, string $template, array $data = [])
+    {
+        $this->email = Injector::inst()->create(Email::class . '.MFA.Notification')
+            ->setTo($member->Email)
+            ->setHTMLTemplate($template)
+            ->setData(array_merge(['Member' => $member], $data));
+        foreach (['to', 'from', 'subject'] as $header) {
+            if (isset($data[$header])) {
+                $method = 'set' . ucwords($header);
+                $this->email->$method($data[$header]);
+            }
+        }
+    }
+
+    /**
+     * Sends the notification to the member
+     *
+     * @return void
+     */
+    public function send()
+    {
+        $this->extend('onBeforeSend', $this->email);
+        try {
+            $this->email->send();
+        } catch (Exception $e) {
+            Injector::inst()->get(LoggerInterface::class . '.MFA')->info($e->getMessage());
```

Comment:
I think it'd be good to put this into DI configuration if you can. Also, do you want an `onAfterSend` hook as well? Currently you could put some logic into `onBeforeSend` that would fire even if the email send fails, but you may want to hook into it and check the result of the send operation

Reply:
I added the after hook, but am not sure what you mean by the DI configuration part.
We already have a DI configuration for MFA logging, however I did have the case wrong on the context (now fixed).

Reply:
Here's an example:

```php
protected $logger;
private static $dependencies = ['Logger' => '%$' . LoggerInterface::class . '.mfa'];

public function setLogger(LoggerInterface $logger) { $this->logger = $logger; return $this; }

// ...

} catch (Exception $e) {
    $this->logger->info($e->getMessage());
}
```

Reply:
Did this.